File: Easy/add_binary/add_binary.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution {
    public:
        string addBinary(string a, string b) {
            string result = "";
            int carry = 0;
            int i = a.length() - 1;
            int j = b.length() - 1;

            while (i >= 0 || j >= 0 || carry) {
                int sum = carry;

                if (i >= 0) {
                    sum += a[i] - '0';
                    i--;
                }

                if (j >= 0) {
                    sum += b[j] - '0';
                    j--;
                
                }
                carry = sum / 2;
                result = to_string(sum % 2) + result;
            }

            return result;
        }
};

//Example usage:
int main() {
    Solution solution1;
    string a = "11";
    string b = "1";
    cout << "Sum of \"" << a << "\" and \"" << b << "\" is: \"" << solution1.addBinary(a, b) << "\"" << endl;

    Solution solution2;
    a = "1010";
    b = "1011";
    cout << "Sum of \"" << a << "\" and \"" << b << "\" is: \"" << solution2.addBinary(a, b) << "\"" << endl;
}

File: Easy/add_binary/add_binary.py
from typing import List

def addBinary(a: str, b: str) -> str:
    i = len(a) - 1
    j = len(b) - 1
    carry = 0
    result = ''

    while i >= 0 or j >= 0 or carry:
        total = carry
        if i >= 0:
            total += int(a[i])
            i -= 1
        if j >= 0:
            total += int(b[j])
            j -= 1

        carry = total // 2
        result = str(total % 2) + result

    return result

if __name__ == "__main__":
    print("Testing addBinary function:")
    print("For the input a = '11', b = '1', the output is:", addBinary('11', '1')) #Output: '100'
    print("For the input a = '1010', b = '1011', the output is:", addBinary('1010', '1011')) #Output: '10101'
    